crates/sequencer/src/sequencer.rs

```diff
@@ -43,6 +49,24 @@ where
     }
 }
 
+impl<S> Sequencer<S>
+where
+    S: DumpLoad + Clone,
+    for<'any> &'any mut S: State + StateReader,
```

Comment:
why do we need this higher rank lifetime?

Reply:
It's required by the Sequencer